x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecksTests.java

```diff
@@ -1653,4 +1662,75 @@ public void testComponentTemplateWithBoostedFieldDynamicTemplate() throws IOExce
         );
         assertEquals(singletonList(expected), issues);
     }
+
+    public void testEmptyDataTierPreference() {
+        List<String> indexNames = new ArrayList<>();
+        indexNames.add(".some_dotted_index");
+        for (int i = 0; i < 15; i++) {
+            indexNames.add("test-index-name-" + i);
+        }
+
+        List<IndexMetadata> indices = new ArrayList<>();
+        for (String indexName : indexNames) {
+            indices.add(
+                IndexMetadata.builder(indexName)
+                    .settings(settings(Version.CURRENT).put(DataTier.TIER_PREFERENCE_SETTING.getKey(), "  "))
+                    .numberOfShards(randomIntBetween(1, 100))
+                    .numberOfReplicas(randomIntBetween(1, 100))
+                    .build()
+            );
+        }
+
+        {
+            List<DeprecationIssue> issues = DeprecationChecks.filterChecks(CLUSTER_SETTINGS_CHECKS, c -> c.apply(ClusterState.EMPTY_STATE));
+            assertThat(issues, empty());
+        }
+
+        {
+            Metadata.Builder metadata = Metadata.builder();
+            for (IndexMetadata indexMetadata : indices) {
+                metadata.put(indexMetadata, false);
+            }
+            ClusterState clusterState = ClusterState.builder(clusterStateWithoutAllDataRoles()).metadata(metadata).build();
+
+            List<DeprecationIssue> issues = DeprecationChecks.filterChecks(CLUSTER_SETTINGS_CHECKS, c -> c.apply(clusterState));
+            assertThat(
+                issues,
+                contains(
+                    new DeprecationIssue(
+                        DeprecationIssue.Level.WARNING,
+                        "No [index.routing.allocation.include._tier_preference] is set for indices [test-index-name-0, "
```

Comment:
Could someone from the docs team review this message? I would expect a list like this to be structured more like:

> No index.routing.allocation.include._tier_preference index setting is set for indices test-index-name-1, test-index-name-10, test-index-name-11, and 13 more.

The main changes being:
* Removal of formatting that can't be communicated in speech which in this case is square brackets. I understand this might mean inconsistency with messages elsewhere so this might not be worth changing for this release.
* Listing fewer indices since long lists are difficult to read if they're presented in sentence form.
* Simplifying the "and more" messaging to indicate how many are unlisted, rather than providing a total count vs. omitted count.

Because folks might want to know which specific indices are being omitted, I also suggest providing a reference to an API that can be used to retrieve the full list. For example:

> To see the full list of indices, use the Get Indices API with the flimborkle parameter set to "stun".

Reply:
@cjcenizal - I think the format of the message here is a bit incidental.  Index names can be quite large and we could spend a considerable amount of time shaping this message to handle multiple cases for readability. However, I don't think there are any easy wins here and don't think time spent would set us up much better for future cases either. Ideally these messages would also be i18n and the UI would own how they are displayed.

As-is we are using a common utility (usually for logs) to format the string and I think in this case that good enough here. Are you OK if we leave this as-is ?

Reply:
Jake and I chatted and he explained to me that the logging utility doesn't offer enough flexibility to implement my suggestion, so I'm happy to see this merged with the current message format.

In the long-term, we agree that good copy is essential to good UX and we'll have to figure out a way to implement that in this and similar cases.